core/src/main/java/com/linecorp/armeria/common/stream/NeverInvokedSubscriber.java

```diff
@@ -30,21 +34,22 @@
 
     @Override
     public void onSubscribe(Subscription s) {
-        throw new IllegalStateException("onSubscribe(" + s + ')');
+        s.cancel();
+        logger.warn("onSubscribe({}) is invoked.", s);
```

Comment:
I would say none of these methods should be invoked. The exceptions that were thrown in this classes are like an assertion, i.e. better leaving as is?

Reply:
Sounds good. Let me revert this class. 😉